Market Tools/macd_strategy.py

```diff
@@ -0,0 +1,147 @@
+# Import the required libraries
+import numpy as np
+import pandas as pd
+import hvplot.pandas
+from pathlib import Path
+from datetime import date
+from pandas.core.frame import DataFrame
+import yfinance as yf
+import os
+import pricing
+
+def analyze_macd(signals_df:DataFrame):
```

Comment:
[Low priority, not for this PR]:  We were pressed for time, but in the future you will want to issue conditional statements that validate your input parameters....that they are the right type, that they are present, etc. If they aren't valid, then the code should return an exception message.